File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/budget/dto/BudgetRequestRestDTO.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.budget.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
public class BudgetRequestRestDTO {

    private Integer id;
    private String idPatient;
    private List<BudgetLineRestDTO> treatments;

    public BudgetRequestRestDTO() {
        treatments = new ArrayList<>();
    }
}

File: front/dentalium/src/app/model/api/BudgetRequestRestDTO.ts
export class BudgetRequestRestDTO {
  id?: number;
  idPatient: string;
  treatments?: BudgetLineRestDTO[];

  constructor(
    idPatient: string = '',
    treatments: BudgetLineRestDTO[] = []
  ) {
    this.idPatient = idPatient;
    this.treatments = treatments;
  }
}


export class BudgetLineRestDTO {
  treatmentId: number;
  quantity: number;
  teeth: string;
  discount: number;

  constructor(
    treatmentId: number = 0,
    quantity: number = 0,
    teeth: string = '',
    discount: number = 0
  ) {
    this.treatmentId = treatmentId;
    this.quantity = quantity;
    this.teeth = teeth;
    this.discount = discount;
  }
}
